### PR TITLE
[Feature] RLHF Reward Model (reopened)

### PR DESCRIPTION
**Pull request by vmoens**
See https://github.com/pytorch/rl/pull/1315 for discussion

### CODE REVIEW

File: torchrl/modules/models/rlhf.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            # Retrieve first index where trajectories diverge
+            divergence_ind = (chosen_ids[i] != rejected_ids[i]).nonzero()[0]
+
+            # Index into the correct rewards
+            c_truncated_reward = chosen_rewards[i][divergence_ind:end_ind]
+            r_truncated_reward = rejected_rewards[i][divergence_ind:end_ind]
+
+            loss += -torch.log(
+                torch.sigmoid(c_truncated_reward - r_truncated_reward)
+            ).mean()
```
Comment (vmoens) [inline torchrl/modules/models/rlhf.py:102]: @tcbegley use F.logsigmoid instead

File: torchrl/modules/models/rlhf.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    @staticmethod
+    def compute_reward_loss(chosen_batch, rejected_batch, pad_token_id=50256):
+        """Compute the reward loss given a chosen and rejected batch.
+
+        The loss is computed as ``loss = -log_sigmoid(chosen_reward - rejected_reward)``.
+        This loss is small when the reward model favours the chosen data and large if
+        the model favours the rejected data.
+
+          .. note:: The loss is computed excluding the common "prefix" subsequence to effectively disregard contribution of the original prompt.
```
Comment (vmoens) [inline torchrl/modules/models/rlhf.py:74]: add small example

**Review by vmoens — COMMENTED** [review]
LGTM
